src/e2e/java/teammates/e2e/pageobjects/AppPageNew.java

```diff
@@ -0,0 +1,687 @@
+        waitForElementVisibility(By.className("modal-backdrop"));
+    }
+
+    void waitForModalHidden(WebElement modalBackdrop) {
+        // Possible exploration: Change to listening to modal hidden event as
+        // this is based on the implementation detail assumption that once modal-backdrop is removed the modal is hidden
+        waitForElementStaleness(modalBackdrop);
+    }
+
+    /**
+     * Waits for the element to appear in the page, up to the timeout specified.
+     */
+    public WebElement waitForElementPresence(By by) {
+        return waitFor(ExpectedConditions.presenceOfElementLocated(by));
+    }
+
+    public void reloadPage() {
+        browser.driver.get(browser.driver.getCurrentUrl());
+        waitForPageToLoad();
+    }
+
+    protected Object executeScript(String script, Object... args) {
+        JavascriptExecutor javascriptExecutor = (JavascriptExecutor) browser.driver;
+        return javascriptExecutor.executeScript(script, args);
+    }
+
+    /**
+     * Equivalent of student clicking the 'Home' tab on the top menu of the page.
+     * @return the loaded page
+     */
+    public StudentHomePageNew loadStudentHomeTab() {
+        click(studentHomeTab);
+        browser.waitForPageLoad();
+        return changePageType(StudentHomePageNew.class);
+    }
+
+    /**
+     * Equivalent of clicking the 'Profile' tab on the top menu of the page.
+     * @return the loaded page
+     */
+    public StudentProfilePageNew loadStudentProfileTab() {
+        click(studentProfileTab);
+        browser.waitForPageLoad();
+        return changePageType(StudentProfilePageNew.class);
+    }
+
+    /**
+     * Equivalent of clicking the 'Help' tab on the top menu of the page.
+     * @return the loaded page
+     */
+    public StudentHelpPageNew loadStudentHelpTab() {
+        click(studentHelpTab);
+        browser.waitForPageLoad();
+        return changePageType(StudentHelpPageNew.class);
+    }
+
+    /**
+     * Click the 'logout' link in the top menu of the page.
+     */
+    public AppPageNew clickLogout() {
+        click(logoutButton);
+        browser.waitForPageLoad();
+        return this;
+    }
+
+    /**
+     * Returns the HTML source of the currently loaded page.
+     * TODO: remove this method as it does not return necessary html anymore since frontend is generated by angular
+     */
+    public String getPageSource() {
+        return browser.driver.getPageSource();
+    }
+
+    public String getTitle() {
+        return browser.driver.getTitle();
+    }
+
+    public void clickLoginDropDown() {
+        click(By.cssSelector("a.nav-link.dropdown-toggle"));
+    }
+
+    public boolean verifyStrangerMessage() {
+        return browser.driver.findElement(By.tagName("h5")).getText()
+                .contains("Your Google account is not known to TEAMMATES");
+    }
+
+    public boolean verifyFirstPanelFeedbackSessionName(String message) {
```

Comment:
This is specific to student home page, so it should not be in this parent class.

Reply:
I was thinking in the future can be applied to instructor home page too since there is panel involved in the same format?

Reply:
Then you're adding a coupling between those two pages.

Reply:
oh ya, my bad! I forgot that they are not sharing the same panel component in the front end.